### PR TITLE
Support parameter tags in optimizer weight decay

### PR DESCRIPTION
**Pull request by dlwh**
## Summary
- allow OptimizerConfig to tag parameters via `TagPattern`
- use parameter tags when computing weight decay mask
- support tag-based learning rate configuration through dict `learning_rate`
- implement `scale_by_tagged_learning_rates` to apply tag-based LRs
- document tagging features in the configuration guide with an expanded section on tag matching
- extend tests for tag priority and multiple tagged learning rates

## Testing
- `pre-commit run --files docs/reference/Configuration.md src/levanter/optim/config.py tests/test_weight_decay_mask.py`
- `pytest tests/test_weight_decay_mask.py -q`

------
https://chatgpt.com/codex/tasks/task_e_685068f5119c8331ae1085b344c65a44

### CODE REVIEW

File: src/levanter/optim/config.py [hunk 1]
```diff
@@ -29,6 +29,22 @@ class LrScheduleContext:
...
+
+    Parameters
+    ----------
+    pattern : str | list[str]
+        Name or names to match against parameter paths or module class names.
+    tag : str
+        Tag to apply to matched parameters.
+    """
+
+    pattern: str | list[str]
```
Comment (dlwh) [inline src/levanter/optim/config.py]:
```suggestion
    pattern: list[str] | str
```
